sdcm/sla/sla_tests.py

```diff
@@ -0,0 +1,152 @@
+                                                         wait_cpu_utilization=self.MIN_CPU_UTILIZATION,
+                                                         prometheus_stats=prometheus_stats)
+                end_time = time.time()
+
+                runtime_per_role_before_increase = self.validate_scheduler_runtime(start_time=start_time,
+                                                                                   end_time=end_time,
+                                                                                   read_users=read_users,
+                                                                                   prometheus_stats=prometheus_stats,
+                                                                                   db_cluster=tester.db_cluster)
+                # TODO: add validation
+                LOGGER.debug('RUN TIME PER ROLE BEFORE INCREASE: {}'.format(runtime_per_role_before_increase))
+            except Exception as details:  # pylint: disable=broad-except
+                wp_event.add_error([str(details)])
+                wp_event.full_traceback = traceback.format_exc()
+                wp_event.severity = Severity.ERROR
+
+    # pylint: disable=too-many-arguments
+    def alter_sl_and_validate_scheduler_runtime(self, tester, service_level, new_shares, read_users, prometheus_stats,
+                                                sleep=600):
+        # pylint: disable=not-context-manager
+        with WorkloadPrioritizationTestEvent(step=f"Alter shares from {service_level.shares} to {new_shares} Service "
+                                                  f"Level {service_level.name} and validate scheduler runtime "
+                                                  f"during load") as wp_event:
+            try:
+                service_level.alter(new_shares=new_shares)
+                start_time = time.time() + 60
+                time.sleep(sleep)
+                end_time = time.time()
+                runtime_per_role_before_increase = self.validate_scheduler_runtime(start_time=start_time,
+                                                                                   end_time=end_time,
+                                                                                   read_users=read_users,
+                                                                                   prometheus_stats=prometheus_stats,
+                                                                                   db_cluster=tester.db_cluster)
+                # TODO: add validation
+                LOGGER.debug('RUN TIME PER ROLE AFTER INCREASE: {}'.format(runtime_per_role_before_increase))
+            except Exception as details:  # pylint: disable=broad-except
+                wp_event.add_error([str(details)])
+                wp_event.full_traceback = traceback.format_exc()
+                wp_event.severity = Severity.ERROR
+
+
+class SlaTests(Steps):
```

Comment:
this class is title as test, but it can be run on it's own so we should call it something appropriate, 
sequence, flow, block, subtests

Reply:
Actually it's whole test. Not sequence, flow, block or subtests

Reply:
a whole test, i.e. this can be run on it's own ? (it doesn't seem to inherit from Tester class)

Reply:
Yes, it is full flow that can be run on prepared dataset